esp/src/eclwatch/ECLSourceWidget.js

```diff
@@ -81,17 +78,29 @@ define([
             if (params.readOnly !== undefined)
                 this.readOnly = params.readOnly;
 
-            this.editor = CodeMirror.fromTextArea(document.getElementById(this.id + "EclCode"), {
-                tabMode: "indent",
-                matchBrackets: true,
-                lineNumbers: true,
-                mode: mode,
-                readOnly: this.readOnly,
-                foldGutter: mode === "xml" ? true : false,
-                gutters: mode === "xml" ? ["CodeMirror-linenumbers", "CodeMirror-foldgutter"] : ["CodeMirror-linenumbers"]
+            var eclContent = dom.byId(this.id + "EclContent");
+            var isVisible = function (domNode) {
+                var bRect = domNode.getBoundingClientRect();
+                return (
+                    bRect.top > 0 && bRect.left > 0 &&
+                    bRect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+                    bRect.left <= (window.innerWidth || document.documentElement.clientWidth)
+                );
+            };
+            var observer = new MutationObserver(mutations => {
+                if (!isVisible(eclContent)) return;
+                this.editor.target(this.id + "EclContent").render();
+                dom.byId(this.id + "EclCode").style.display = "none";
+                observer.disconnect();
             });
-            dom.byId(this.id + "EclContent").style.backgroundColor = this.readOnly ? 0xd0d0d0 : 0xffffff;
-            this.editor.setSize("100%", "100%");
+            observer.observe(eclContent, { attributes: true, attributeFilter: ["style"] });
+
+            this.editor = new CodeMirror.ECLEditor();
+            eclContent.style.backgroundColor = this.readOnly ? 0xd0d0d0 : 0xffffff;
+            // force a style update, for some pages loading this widget
+            // which don't "naturally" cause a style change during init
+            eclContent.style.paddingTop = "1px";
+            eclContent.style.paddingTop = "0px";
```

Comment:
If the initialization of the code mirror widget was relocated to "init" and extra code for resizing was added to the resize method, I suspect we wouldn't need the observable?

Reply:
Alternatively we could wrap this into a Utility function also?

Reply:
Move the mutation observer to a more generic function in Utility. The initialization of the CM widget does seem to already be located in init.